src/io/memory.cr

```diff
@@ -225,7 +225,9 @@ class IO::Memory
     if pos == @bytesize
       ""
     else
-      String.new(@buffer + @pos, @bytesize - @pos)
+      String.new(@buffer + @pos, @bytesize - @pos).tap {
```

Comment:
The ruby style guide discourages it for the good reason that it encourages chaining methods on the return values of multi-line blocks. And I tend to agree with it. I'm not sure what the core team thinks though.